### PR TITLE
Feat/vote disable

### LINKED ISSUE #248: [feat] Disable vote btn
### This issue will be part of ODHack10, please apply via Onlydust app
- If the user already voted to the funding, disable the vote btn for him.
- There is a getVoter function in the backend, you can use that function to achieve this.
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue.

### PR DESCRIPTION
**Pull request by kimcascante**
# Pull Request

- [x] Closes #248
- [ ] Added tests (if necessary)
- [x] Run tests
- [x] Run formatting
- [x] Commented the code

## Changes description

Describe what your changes will do and any TODOs there might be in the future related to the issue you are closing.
Adding the function in FundVote.tsx to disable btn when a user already voted, using the getVoter function in fund.cairo in the backend, also showing the btn when is checking if user has voted.

## Current output

Give evidence like screenshots of what your job looks like.

Before voting:
Captura de pantalla 2024-11-27 a la(s) 9 02 15 a  m

Reviewing if user has voted:
Captura de pantalla 2024-11-27 a la(s) 9 04 45 a  m

After voting:
Captura de pantalla 2024-11-27 a la(s) 6 42 19 a  m

## Time spent breakdown

Give a breakdown on the time you used to solve the issue you are closing.

Backend and Frontend Setup 2 hours 
Understanding the code project 1 day
Testing the vote funtionality 1 day
Solution and implementation 2 days
Testing the implementation 1 day

## Comments

Any other details you might want to share.

Was difficult to me to test at the beginning because didn't use wallets before so when I was trying to test the vote funtionality I had the error '''Tx not executed:Transaction failure predicted'' so the error was that I was using a different browser where I had the sepolia network. 

I tried to keep the compatibility and previous implementation.

### CODE REVIEW

File: frontend/gostarkme-web/.env.example
Comment (adrianvrj) [inline frontend/gostarkme-web/.env.example:1]: please revert the changes to this file, you deleted it 😅

File: frontend/gostarkme-web/components/modules/Fund/FundVote.tsx [hunk 1]
```diff
@@ -7,16 +7,18 @@ import { latestTxAtom } from "@/state/latestTx";
 import { useAtomValue, useSetAtom } from "jotai";
 import { Contract, InvokeFunctionResponse } from "starknet";
 import { useRouter } from "next/navigation";
+import {useState, useEffect} from "react"; //import useState and useEffect
```
Comment (adrianvrj) [inline frontend/gostarkme-web/components/modules/Fund/FundVote.tsx:10]: delete this comment here please

File: frontend/gostarkme-web/next.config.mjs
Comment (adrianvrj) [inline frontend/gostarkme-web/next.config.mjs:1]: please revert the changes to this file too

**Review by adrianvrj — CHANGES_REQUESTED** [review]
everything seems fine, could you update it in order to check the status when the funding page is rendering rather than when the user clicks on the vote btn?

File: frontend/gostarkme-web/.env.example
Comment (kimcascante) [reply frontend/gostarkme-web/.env.example:1]: Hi @adrianvrj in the [Local configurations ](https://github.com/kimcascante/gostarkme/blob/dev/frontend/README.md) it says to rename and comment next.config.mjs, so it should be revert it before the PR?

File: frontend/gostarkme-web/components/modules/Fund/FundVote.tsx (hunk 1 above)
Comment (kimcascante) [reply frontend/gostarkme-web/components/modules/Fund/FundVote.tsx:10]: I am deleting it :)

File: frontend/gostarkme-web/.env.example
Comment (adrianvrj) [reply frontend/gostarkme-web/.env.example:1]: Yes, it is only for local config, if we upload this to a production env the changes would make the build fail

File: frontend/gostarkme-web/next.config.mjs
Comment (kimcascante) [reply frontend/gostarkme-web/next.config.mjs:1]: @adrianvrj updated! can you please review again and confirm? should work like this 

https://github.com/user-attachments/assets/7c3c7dd9-a48d-47f2-b289-1522b084fb5e

Thank you for your time

**Review by adrianvrj — APPROVED** [review]

**Comment by adrianvrj** [comment]
@kimcascante there are some conflicts, could you please resolve them in order to merge your pr? :)

**Comment by kimcascante** [comment]
> @kimcascante there are some conflicts, could you please resolve them in order to merge your pr? :)

No conflicts now. :), and thank you

**Review by adrianvrj — APPROVED** [review]
